Remotion/Development/Core/UnitTesting/UnloadableAssemblyContext.cs

```diff
@@ -0,0 +1,98 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+#if !NETFRAMEWORK
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Loader;
+using System.Threading;
+
+namespace Remotion.Development.UnitTesting
+{
+  /// <summary>
+  /// Wraps an unloadable <see cref="AssemblyLoadContext"/> with the dispose pattern.
+  /// </summary>
+  public class UnloadableAssemblyContext : IDisposable
```

Comment:
`AssemblyLoadContext` does not implement `IDisposable`. It can be marked as "unloadable" which can then be triggered using `.Unload()`. The thing is that it just removes the "keep alive" reference to the `AssemblyLoadContext` but does not wait for the actual unload. This class is supposed to provide that with the using pattern.

Reply:
so, better summary it is, then. "AssemlyLoadContext that waits until the unload operation has completed" or something similiar/better.